diff --git a/include/fmt/format.h b/include/fmt/format.h
index 782cf9b..55e892b 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -312,7 +312,7 @@ typename Allocator::value_type* allocate(Allocator& alloc, std::size_t n) {
 FMT_END_NAMESPACE
 
 namespace std {
-using namespace fmt::v5::internal;
+// using namespace fmt::v5::internal;
 // Standard permits specialization of std::numeric_limits. This specialization
 // is used to detect presence of uintptr_t.
 template <>
